app/src/main/java/fr/free/nrw/commons/utils/SystemThemeUtils.java

```diff
@@ -0,0 +1,45 @@
+package fr.free.nrw.commons.utils;
+
+import android.content.Context;
+import android.content.res.Configuration;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import fr.free.nrw.commons.kvstore.JsonKvStore;
+import fr.free.nrw.commons.settings.Prefs;
+
+public class SystemThemeUtils {
+
+    private Context context;
+    private JsonKvStore applicationKvStore;
+
+    @Inject
+    public SystemThemeUtils(Context context, @Named("default_preferences") JsonKvStore applicationKvStore) {
+        this.context = context;
+        this.applicationKvStore = applicationKvStore;
+    }
+
+    // Return true is system wide dark theme is enabled else false
+    public boolean getSystemDefaultThemeBool(String theme) {
+        if (theme.equals("1")) {
```

Comment:
These are magic strings and it would be good if we could not be using them like this.

There are 2 options:

1. extract constants for `1`,`2`,`3` `THEME_MODE_DEFAULT` etc
1. create a enum to represent these values, wrapping the retrieval from the keystore

I much prefer 2 because it is a superiorily typed expression. 2 is easier. Your choice

Reply:
Done!